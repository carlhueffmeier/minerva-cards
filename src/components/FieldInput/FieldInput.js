import React from 'react';
import PropTypes from 'prop-types';

FieldInput.propTypes = {
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
  values: PropTypes.object.isRequired,
  onFieldChange: PropTypes.func.isRequired
};

function FieldInput(props) {
  return (
    <div>
      {props.fields.map(fieldId => (
        <label key={fieldId}>
          {fieldId}
          <input
            value={props.values[fieldId] || ``}
            onChange={event => props.onFieldChange(fieldId, event.target.value)}
          />
        </label>
      ))}
    </div>
  );
}

export default FieldInput;
