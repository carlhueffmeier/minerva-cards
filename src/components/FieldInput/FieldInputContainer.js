import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { selectors as typeSelectors } from 'state/modules/types';
import FieldInput from './FieldInput';

FieldInputContainer.propTypes = {
  typeId: PropTypes.string.isRequired,
  type: PropTypes.object.isRequired,
  values: PropTypes.object.isRequired,
  onFieldChange: PropTypes.func.isRequired
};

function FieldInputContainer(props) {
  const { type = {}, values } = props;
  return (
    <FieldInput
      fields={type.fields || {}}
      values={values}
      onFieldChange={props.onFieldChange}
    />
  );
}

function mapStateToProps(state, ownProps) {
  const { typeId } = ownProps;
  return {
    type: typeSelectors.getTypeById(state, typeId)
  };
}

export default connect(mapStateToProps, null)(FieldInputContainer);
