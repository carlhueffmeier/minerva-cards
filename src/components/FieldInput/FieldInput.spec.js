import React from 'react';
import { shallow } from 'enzyme';
import FieldInput from './FieldInput';

describe('FieldInput', () => {
  const props = {
    fields: [`Kanji`, `Reading`, `Examples`],
    values: {
      Kanji: `犬`,
      Reading: `いぬ`,
      Examples: `賢い犬だな。`
    },
    onFieldChange: () => null
  };

  const wrapper = shallow(<FieldInput {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
