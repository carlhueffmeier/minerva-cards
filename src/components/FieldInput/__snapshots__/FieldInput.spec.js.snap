// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldInput
    fields={
        Array [
            "Kanji",
            "Reading",
            "Examples",
          ]
    }
    onFieldChange={[Function]}
    values={
        Object {
            "Examples": "賢い犬だな。",
            "Kanji": "犬",
            "Reading": "いぬ",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Kanji
          <input
                    onChange={[Function]}
                    value="犬"
          />
</label>,
        <label>
          Reading
          <input
                    onChange={[Function]}
                    value="いぬ"
          />
</label>,
        <label>
          Examples
          <input
                    onChange={[Function]}
                    value="賢い犬だな。"
          />
</label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Kanji",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Kanji",
            <input
              onChange={[Function]}
              value="犬"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Kanji",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "value": "犬",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "Reading",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Reading",
            <input
              onChange={[Function]}
              value="いぬ"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Reading",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "value": "いぬ",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "Examples",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Examples",
            <input
              onChange={[Function]}
              value="賢い犬だな。"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Examples",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "value": "賢い犬だな。",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Kanji
            <input
                        onChange={[Function]}
                        value="犬"
            />
</label>,
          <label>
            Reading
            <input
                        onChange={[Function]}
                        value="いぬ"
            />
</label>,
          <label>
            Examples
            <input
                        onChange={[Function]}
                        value="賢い犬だな。"
            />
</label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Kanji",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Kanji",
              <input
                onChange={[Function]}
                value="犬"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Kanji",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "value": "犬",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "Reading",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Reading",
              <input
                onChange={[Function]}
                value="いぬ"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Reading",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "value": "いぬ",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "Examples",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Examples",
              <input
                onChange={[Function]}
                value="賢い犬だな。"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Examples",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "value": "賢い犬だな。",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
